src/com/ableton/Pyenv.groovy

```diff
@@ -24,6 +24,13 @@ class Pyenv implements Serializable {
    */
   String pyenvRoot
 
+  Pyenv(Object script, String pythonVersion) {
+    this.script = script
+    this.pythonVersion = pythonVersion
+    assert script.env['PYENV_ROOT']
```

Comment:
This contradicts https://github.com/Ableton/python-pipeline-utils/pull/102#discussion_r970693413, where you wrote "No, we don't assert in constructors."